exp/internal/controllers/machinepool_controller_noderef.go

```diff
@@ -81,8 +83,14 @@ func (r *MachinePoolReconciler) reconcileNodeRefs(ctx context.Context, cluster *
 		return ctrl.Result{}, err
 	}
 
+	// Check nodeRefMapResult in scope for error.
+	if s.nodeRefMapResult.err != nil {
```

Comment:
Once we handle the error in reconcileInfrastructure we can drop this here

Reply:
cool so im still doing a check for nil here and returning early -- let me know if you think there's a more appropriate way to handle

Reply:
bumping this when you have a chance to take another pass @sbueringer 🙇

Reply:
That's okay